### PR TITLE
Sync table from frontend

### PR DESCRIPTION
**Pull request by thomtrp**
This PR:
- separates the existing updateSyncStatus endpoint into 2 endpoints
- creates mutations and hooks that will call those endpoints
- trigger the hook on toggle
- removes form logic and add a separated component for toggling

### CODE REVIEW

File: packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx [hunk 1]
```diff
@@ -25,27 +28,63 @@ const StyledRowRightContainer = styled.div`
...
-    useFormContext<SettingsIntegrationsDatabaseTablesFormValues>();
+  const { syncRemoteTable } = useSyncRemoteTable();
+  const { unsyncRemoteTable } = useUnsyncRemoteTable();
 
+  const [items] = useState(
+    tables.map((table) => ({
+      id: table.name,
+      ...table,
+    })),
+  );
```
Comment (thaisguigon) [inline packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx:42]: Maybe add a comment on why we need a `useState`?

File: packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx [hunk 2]
```diff
@@ -25,27 +28,63 @@ const StyledRowRightContainer = styled.div`
...
+
+  const onSyncUpdate = useCallback(
+    async (value: boolean, tableName: string) => {
+      const table = items.filter((table) => table.name === tableName)[0];
```
Comment (thaisguigon) [inline packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx:46]:
```suggestion
      const table = items.find((table) => table.name === tableName);
```

File: packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx [hunk 3]
```diff
@@ -25,27 +28,63 @@ const StyledRowRightContainer = styled.div`
...
+
+      // eslint-disable-next-line @nx/workspace-explicit-boolean-predicates-in-if
+      if (!table) return;
```
Comment (thaisguigon) [inline packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx:49]:
```suggestion
      if (!isDefined(table)) return;
```

File: packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationSyncStatusToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react';
+
+import { Toggle } from '@/ui/input/components/Toggle';
+import { RemoteTableStatus } from '~/generated-metadata/graphql';
+
+export const SettingsIntegrationSyncStatusToggle = ({
+  item: table,
```
Comment (thaisguigon) [inline packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationSyncStatusToggle.tsx:7]: This component seems to be specific to the RemoteTable settings. The `item` prop can be named `table` directly, and the name of the component could include `RemoteTable` for clarity (`SettingsIntegrationsRemoteTableSyncStatusToggle`).

**Review by thaisguigon — CHANGES_REQUESTED** [review]

**Review by thaisguigon — APPROVED** [review]

File: packages/twenty-front/src/modules/databases/hooks/useSyncRemoteTable.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { useCallback } from 'react';
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/databases/hooks/useSyncRemoteTable.ts:1]: databases modules does not look clear to me. It looks to be part of integrations

File: packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx [hunk 6]
```diff
@@ -25,27 +29,63 @@ const StyledRowRightContainer = styled.div`
...
+  // We need to use a state because the table status update re-render the whole list of toggles
+  const [items] = useState(
+    tables.map((table) => ({
+      id: table.name,
+      ...table,
+    })),
+  );
+
+  const onSyncUpdate = useCallback(
+    async (value: boolean, tableName: string) => {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx:47]: value naming is not clear
syncStatus?

Comment (charlesBochet) [reply packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx:47]: isSynced?

my brain had a hard time understanding if (value) line

File: packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts [hunk 7]
```diff
@@ -261,6 +269,12 @@ export class RemoteTableService {
     await this.workspaceMigrationRunnerService.executeMigrationFromPendingMigrations(
       workspaceId,
     );
+
+    return {
+      name: input.name,
+      schema: input.schema,
+      status: RemoteTableStatus.NOT_SYNCED,
+    } as RemoteTableDTO;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts:277]: can we avoid using  'as' here what's the issue?

File: packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts [hunk 8]
```diff
@@ -220,12 +222,18 @@ export class RemoteTableService {
         });
       }
     }
+
+    return {
+      name: input.name,
+      schema: input.schema,
+      status: RemoteTableStatus.SYNCED,
+    } as RemoteTableDTO;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts:230]: same here!

**Review by charlesBochet — COMMENTED** [review]
LGTM with minor comments!
Tested it, this feature is awesome

**Review by charlesBochet — APPROVED** [review]

File: packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx (hunk 6 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/settings/integrations/components/SettingsIntegrationDatabaseTablesListCard.tsx:47]: isSynced 👍

File: packages/twenty-front/src/modules/databases/hooks/useSyncRemoteTable.ts (hunk 5 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/databases/hooks/useSyncRemoteTable.ts:1]: looks like a global change that we need to discuss with @thaisguigon! No really related to that PR

**Pull request merged**